enhancements/ingress/lb-subnet-selection-aws.md

```diff
@@ -0,0 +1,597 @@
+upon updating `Subnets` (see [Effectuating Subnet Updates](#effectuating-subnet-updates)).
+See the next section, [Replacing All Subnets in a CLB](#replacing-all-subnets-in-a-clb)
+for more details on #3.
+
+##### Replacing All Subnets in a CLB
+
+If an update to the list of subnets in `service.beta.kubernetes.io/aws-load-balancer-subnets`
+would completely replace the existing subnets on an existing service
+using a CLB, the AWS CCM will throw an error and not reconcile the service.
+As outlined in this [GitHub comment](https://github.com/kubernetes/cloud-provider-aws/issues/249#issuecomment-906595623),
+a new set of subnets causes the AWS controller attempts to detach all
+existing subnets on CLBs which is not a supported operation.
+
+A cluster admin wanting to replace all subnets would need to delete and
+recreate the LoadBalancer-type service. It may not be obvious that to the cluster
+admin that they are replacing all subnets because they must be aware of which
+subnets are currently used by the load balancer. This includes automatically
+selected subnets when NOT using `service.beta.kubernetes.io/aws-load-balancer-subnets`.
+
+As mentioned above, to mitigate this we are requiring services to be recreated
+upon updating `Subnets` as outlined in [Effectuating Subnet Updates](#effectuating-subnet-updates).
+
+#### CCM Doesn't Reconcile NLB Subnets Updates
+
+As long as the subnets in the annotation are valid, CLBs can update subnets
+after creation. However, the same is not true for NLBs. Once a NLB is created
+by the CCM, any updates to the `service.beta.kubernetes.io/aws-load-balancer-subnets`
+annotation will be validated, but annotation will NOT impact the subnets on
+the NLB. The [ensureLoadBalancerv2](https://github.com/openshift/cloud-provider-aws/blob/51180c32ceb70298faad1aac1022d8983fdf2e78/pkg/providers/v1/aws_loadbalancer.go#L146)
+function responsible for provisioning NLBs lacks the logic to update NLB
+subnets after provisioning.
+
+Requiring services to be recreated upon updating `Subnets` as outlined in
+[Effectuating Subnet Updates](#effectuating-subnet-updates) will mitigate
+cluster admins from experiencing this issue.
+
+### Drawbacks
+
+This enhancement brings additional engineering complexity for upgrade
+scenarios because cluster admins have previously been allowed to directly
+add this annotation on a service. 
+
+Debugging invalid subnets will be confusing for cluster admins and may
+lead to extra support cases or bugs. By requiring service to be recreated,
+it guarantees that the service won't be provisioned when invalid subnets
+are provided, thus clearly indicating to the cluster admin that something
+is wrong.
+
+The design in [Effectuating Subnet Updates](#effectuating-subnet-updates)
+requires a cluster admin to check the IngressController's status after
+updating `Subnets` for instructions on how to proceed.
+
+Requiring the cluster admin to delete and recreate the LoadBalancer-type
```

Comment:
I would put this at the top, it is the worst drawback.
```suggestion
Requiring the cluster admin to delete the LoadBalancer-type
```

Reply:
Yes I agree. Done.